### DISCUSSION TITLE
Could Apple Watch pushes data remotely without iPhone app?

### DISCUSSION THREAD

**Discussion by XSAM** [tszheichoi/awesome-sensor-logger#46] [General]
Could Apple Watch push data remotely via MQTT without opening the iPhone app? It would be very helpful to use Apple Watch itself to control something.

**Comment by tszheichoi** [comment]
Currently, this is not possible -- the iPhone app needs to be nearby. Are you specifically interested in MQTT push directly from the watch, using its own internet, without the phone nearby?

**Reply by XSAM** [reply]
> Are you specifically interested in MQTT push directly from the watch, using its own internet, without the phone nearby?

Yes. This is what I want.

The workflow could be

- I use Apple Watch's double-tape gesture to open the Sensor Logger app, which then starts recording or pushing data (mostly motion data) to MQTT via the network automatically without the need for a phone nearby.
- Moving around to control something...
- Another double-tape will be used to end the recording (I'm not sure if this is possible).

**Reply by tszheichoi** [reply]
I can certainly integrate with the double tap gesture to start and end recordings (I have added this to my feature roadmap now) But the MQTT still requires the phone to be nearby. There are a lot of challenges to getting MQTT (and for that matter, many other advanced features of Sensior Logger such as Rule Engine) to work standalone on the watch. I will keep this in mind, but it will unlikely happen in the near future.
